eva_submission/submission_qc_checks.py

```diff
@@ -240,64 +252,113 @@ def check_if_accessioning_completed_successfully(self, vcf_files):
+        analysis_to_file_names = {}
+        for analysis_alias, analysis_accession in self.eload_cfg.query('brokering', 'ena', 'ANALYSIS').items():
+            # Find the files associated with this analysis
+            analysis_to_file_names[analysis_accession] = [
+                os.path.basename(f) for f in self.analyses.get(analysis_alias).get('vcf_files')
+            ]
+
+            self._check_if_one_job_completed_successfully(
+                analysis_accession, f"annotation.*{analysis_accession}*.log", "annotate_variants", failed_analysis
+            )
+            self._check_if_one_job_completed_successfully(
+                analysis_accession, f"statistics.*{analysis_accession}*.log", "calculate_statistics", failed_analysis
+            )
+        self._annotation_job_check_result = "PASS"
+        self._statistics_job_check_result = "PASS"
+        if failed_analysis:
+            for analysis_accession in list(failed_analysis):
+                errors = failed_analysis[analysis_accession]
+                # Check that the variant_load step didn't run the annotation and calculate statistics
+                variant_load_error = any(
+                    'variant_load' in failed_files.get(f, {}) for f in analysis_to_file_names[analysis_accession]
+                )
+                if 'annotate_variants' in errors and variant_load_error:
+                    self._annotation_job_check_result = "FAIL"
+                elif 'annotate_variants' in errors and not variant_load_error:
+                    # We can remove the annotate_variants error because it is covered by variant_load
+                    errors.pop('annotate_variants')
+                if 'calculate_statistics' in errors and variant_load_error:
+                    self._statistics_job_check_result = "FAIL"
+                elif 'calculate_statistics' in errors and not variant_load_error:
+                    # We can remove the calculate_statistics error because it is covered by variant_load
+                    errors.pop('calculate_statistics')
+                if not errors:
+                    # If there are no more error we can remove the analysis completely
+                    failed_analysis.pop(analysis_accession)
 
         report = f"""
-                variant load result: {self._variant_load_job_check_result}
+                vcf load result: {self._load_vcf_job_check_result}
+                annotation result: {self._annotation_job_check_result}
+                statistics result: {self._statistics_job_check_result}
                 accession import result: {self._acc_import_job_check_result}"""
         if failed_files:
+            # For the report the variant_load does not needs to be reported because any new run will be done
+            # with the new load_vcf method. Remove the variant_load from the failed files
+            for file_name in failed_files:
+                failed_files[file_name].pop('variant_load', None)
```

Comment:
Will the `variant_load` and `load_vcf` entries technically have different report strings sometimes? For example `variant_load` failed on a particular step but `load_vcf` just says it couldn't find the log?
I guess it might not matter since as you say the new runs will use `load_vcf` either way, but it might be confusing for the user.

Reply:
I agree that it would be confusing for the user: 
It will report that `load_vcf` is missing and subsequently that `annotate_variants` and `calculate_statistics` is also missing simply because `variant_load` did not complete successfully. And in most case the `load_vcf` has been run successfully. 
I did that to avoid making the code overcomplicated to support backward compatibility (it already is to some extent). 
In this scenario we would waste resources rerunning the new version that would provide more clarity of the success and failure.